### PR TITLE
CSHARP-5326: Mark API as obsolete: Guids

### PR DESCRIPTION
**Pull request by papafe**

### CODE REVIEW

**Comment by papafe** [comment]
@BorisDog it seems that most of the things mentioned in the deprecation mentioned have already been done, so this is a very small PR

File: src/MongoDB.Bson/ObjectModel/GuidRepresentationMode.cs [hunk 1]
```diff
@@ -13,12 +13,15 @@
...
 
+using System;
+
 namespace MongoDB.Bson
 {
     /// <summary>
     /// Whether to handle GuidRepresentation using the v2.x mode of the v3.x mode.
     /// See the reference documentation for details.
     /// </summary>
+    [Obsolete("This enum will be made removed in a later release, with V3 being the only representation mode.")]
```
Comment (BorisDog) [inline src/MongoDB.Bson/ObjectModel/GuidRepresentationMode.cs:24]: Please make sure that compilation succeeds before merging

**Review by BorisDog — APPROVED** [review]
LGTM

File: src/MongoDB.Bson/ObjectModel/GuidRepresentationMode.cs (hunk 1 above)
Comment (papafe) [reply src/MongoDB.Bson/ObjectModel/GuidRepresentationMode.cs:24]: Sure

**Pull request merged**